### PR TITLE
Update references to Jira to GH for the Java SDK

### PR DESCRIPTION
**Pull request by damccorm**
Part of https://github.com/apache/beam/issues/21812

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by damccorm** [comment]
Run Java_PVR_Flink_Batch PreCommit

**Comment by damccorm** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by damccorm** [comment]
Run Kotlin_Examples PreCommit

**Comment by damccorm** [comment]
Run Java PreCommit

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokersTest.java [hunk 1]
```diff
@@ -93,7 +93,8 @@
 @SuppressWarnings({
   "rawtypes", // TODO(https://github.com/apache/beam/issues/20447)
   "SameNameButDifferent",
-  "unused" // TODO(BEAM-13271): Remove when new version of errorprone is released (2.11.0)
+  "unused" // TODO(https://github.com/apache/beam/issues/21230): Remove when new version of
+  // errorprone is released (2.11.0)
```
Comment (TheNeuralBit) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokersTest.java:97]: These look weird, any chance your automation could move these comments to the line above?

**Review by TheNeuralBit — APPROVED** [review]

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokersTest.java (hunk 1 above)
Comment (damccorm) [reply sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokersTest.java:97]: I think this was actually caused by the autoformatter (spotlessApply), not my automation itself. With that said, I manually went through and cleaned it up (and some other weird formatting things)

**Comment by damccorm** [comment]
Run Python_PVR_Flink PreCommit

**Comment by damccorm** [comment]
Run Java_PVR_Flink_Docker PreCommit

**Comment by damccorm** [comment]
Run Python_PVR_Flink PreCommit

**Comment by damccorm** [comment]
Run Python_PVR_Flink PreCommit

**Comment by damccorm** [comment]
Run Kotlin_Examples PreCommit

**Comment by damccorm** [comment]
Run Java_PVR_Flink_Docker PreCommit

**Comment by damccorm** [comment]
Run Java PreCommit

**Comment by damccorm** [comment]
Run Spotless PreCommit

**Comment by damccorm** [comment]
Run Java_PVR_Flink_Batch PreCommit

**Comment by damccorm** [comment]
Run Java PreCommit

**Comment by damccorm** [comment]
Run Java PreCommit

**Comment by damccorm** [comment]
Run Java_PVR_Flink_Docker PreCommit

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokersTest.java (hunk 1 above)
Comment (TheNeuralBit) [reply sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokersTest.java:97]: Thanks!

**Pull request merged**